src/data-workspace/category-combo-table/calculate-totals.js

```diff
@@ -0,0 +1,14 @@
+export const calculateColumnTotals = (matrix) =>
+    matrix.length < 2
+        ? matrix[0].map((v) => Number(v) || 0)
+        : matrix.reduce((totals, currRow) =>
+              totals.map((currTotal, i) => sum(currTotal, currRow[i]))
+          )
+
+const sum = (acc, curr) => {
+    const a = Number(curr) || 0
+    const b = Number(acc) || 0
+    return a + b
+}
```

Comment:
Could do this inline as well. But I don't really understand why you would prefer to use a lodash-function when it's so simple?

I'm not sure I follow, notice that this function returns an array... It does not help to provide a default value here, and actually it's deliberately not done because then it uses the first value of the array.
It returns an array of totals. Because it calculates all the sum for all column**s**.

Reply:
I agree that providing an initial value of `0` to `reduce` would be better, because then `sum` could be simplified to this.
```js
const sum = (acc, curr) => acc + (Number(curr) || 0)
```
(at which point you could even consider just using an anonymous arrow function instead of a named const.

I don't think we need to add another dependency to the project for this though.

Reply:
This would need a change to the reduce-function though? Notice how I'm taking advantage of how `reduce` uses the first value of the array if not default is provided. And the first value is an array... `totals` is not a number! It's an array.

Reply:
> Notice how I'm taking advantage of how `reduce` uses the first value of the array as the accumulator if not default is provided.

No I had not noticed that. I personally always like to provide initial values to reduce functions, but I guess there is nothing wrong with doing things this way.

Beyond that I still don't really understand why we need this conditional logic. When we talk about a matrix in this context, I am assuming we are simply talking about an array, which contains a bunch of items, which are also arrays with an equal length. So something like this:
```js
[
    [1,2,3,5],
    [1,2,3,5],
    [1,2,3,5],
]
```

And if that's right then the calculations are like also pretty basic:
```js
const rowTotals = (matrix, row = 0) => matrix[row].reduce((acc, currVal) => acc + (Number(currVal) || 0), 0)
const colTotals = (matrix, column = 0) => matrix.reduce((acc, currRow) => acc + (Number(currRow[column]) || 0), 0)
```

What is more complex about what we are doing?

Reply:
Sorry, I missed the fact that you were not passing a column index to the `calculateColumnTotals` function. So you want the output to be an array. This also made me understand why you want to not provide an initial value, and let the function use the first matrix row instead. But I still don't see the need for conditional logic. The code below should work:
```js
const calculateRowTotals = (matrix, row = 0) =>
    matrix[row].reduce((total, currVal) => total + (Number(currVal) || 0), 0)

const calculateColumnTotals = (matrix) => matrix.reduce((totals, row) => {
    row.forEach((currVal, index) => {
        totals[index] = totals[index] + (Number(currVal) || 0)
    })
    return totals
})
```

Reply:
> `const colTotals = (matrix, column = 0) => 
    matrix.reduce((acc, currRow) => acc + (Number(currRow[column]) || 0), 0)`

Here you are changing the function though. The way the `ColumnTotals`-component works is that it renders the totals for all columns, and thus this function calculates all totals in one go. For `RowTotal` this is different, because they are rendered in-line for each row - and because you are already mapping over rows (when rendering the table), it's easier for that function to just calculate per-row.
In your function you would need an additional map to map over the columns. Then you would end up with something like

```javascript
const calculateColumnTotals = (matrix) => matrix[0].map((_c, i) => colTotals(matrix, i))
```
Which is fine, but I don't really see how that's so much better than the current solution. Guess it might be easier to read, and then you wouldn't need the conditional to check for matrix.length of one.

Reply:
Hmm... Seems like you are right, but I did end up in a situation where I got some error that resulted in the conditional. But it might've been that the matrix was completely empty? In that case we can just remove the conditional in the original function. Because then it works there as well, since the problem would be the outer reduce, not the inner map. 

And if I'm not mistaken your solution basically just changes the functional `.map` to an imperative `forEach`, the result should be identical.

Reply:
I think you are correct. And agree we should add some tests for these edge case, as you suggested on Slack.